libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts

```diff
@@ -0,0 +1,159 @@
+
+    if (items.length === 0) {
+      throw new NotFoundException('We could not find any preferences');
+    }
+
+    const workflowPreferences = this.getWorkflowPreferences(items);
+
+    const userPreferences = this.getUserPreferences(items);
+
+    const subscriberGlobalPreferences =
+      this.getSubscriberGlobalPreferences(items);
+
+    const subscriberWorkflowPreferences = this.getSubscriberWorkflowPreferences(
+      items,
+      command.templateId,
+    );
+
+    /*
+     * Order is important here because we like the workflowPreferences (that comes from the bridge)
+     * to be overridden by any other preferences and then we have preferences defined in dashboard and
+     * then subscribers global preferences and the once that should be used if it says other then anything before it
+     * we use subscribers workflow preferences
+     */
+    const preferences = [
+      workflowPreferences,
+      userPreferences,
+      subscriberGlobalPreferences,
+      subscriberWorkflowPreferences,
+    ]
+      .filter((preference) => preference !== undefined)
+      .map((item) => item.preferences);
+
+    return deepMerge(preferences);
+  }
+
+  public async getPreferenceChannels(command: {
+    environmentId: string;
+    organizationId: string;
+    subscriberId: string;
+    templateId?: string;
+  }): Promise<IPreferenceChannels | undefined> {
+    try {
+      const result = await this.execute(
+        GetPreferencesCommand.create({
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          subscriberId: command.subscriberId,
+          templateId: command.templateId,
+        }),
+      );
+
+      return {
+        in_app:
+          result.channels.in_app.defaultValue || result.workflow.defaultValue,
+        sms: result.channels.sms.defaultValue || result.workflow.defaultValue,
+        email:
+          result.channels.email.defaultValue || result.workflow.defaultValue,
+        push: result.channels.push.defaultValue || result.workflow.defaultValue,
+        chat: result.channels.chat.defaultValue || result.workflow.defaultValue,
+      };
+    } catch (e) {
+      return undefined;
+    }
+  }
+
+  private getSubscriberWorkflowPreferences(
+    items: PreferencesEntity[],
+    templateId: string,
+  ) {
+    return items.find(
+      (item) =>
+        item.type === PreferencesTypeEnum.SUBSCRIBER_WORKFLOW &&
+        item._templateId == templateId,
+    );
+  }
+
+  private getSubscriberGlobalPreferences(items: PreferencesEntity[]) {
+    return items.find(
+      (item) => item.type === PreferencesTypeEnum.SUBSCRIBER_GLOBAL,
+    );
+  }
+
+  private getUserPreferences(items: PreferencesEntity[]) {
+    return items.find(
+      (item) => item.type === PreferencesTypeEnum.USER_WORKFLOW,
+    );
+  }
+
+  private getWorkflowPreferences(items: PreferencesEntity[]) {
+    return items.find(
+      (item) => item.type === PreferencesTypeEnum.CODE_FIRST_WORKFLOW,
+    );
+  }
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** Now that we have these enum values, we could just make a `getPreferenceByType` function that takes the array and the type.

Also, these methods could either be declared as external helper functions, or static methods‏